src/components/TrendingCard/__tests__/TrendingCard.spec.js

```diff
@@ -19,4 +19,20 @@ describe('TrendingCard', () => {
       render(<TrendingCard element={Link} />).container.firstChild
     ).toMatchSnapshot();
   });
+
+  describe('automated accessibility testing', () => {
```

Comment:
I'd like to see if there's a way for us to better integrate these with our snapshot tests, so that we can test a11y against  the same snapshots - For example the default `TrendingCard`, and the custom link variation.

Reply:
I'm having trouble seeing what you are saying...

I thought with this setup, we are already using the story defined in the storybook. And all tests are running through that. I don't know how you'd take a snapshot file's output and run that through AAT? And I wonder even if you should, since they are different? 🤔 Maybe you could post an example of what you mean?

Reply:
Thanks - I understand your confusion! What I think I'm trying to tease out is whether we should have a11y tests separate from the snapshot tests, or if they should be integrated.

For example, this component only has 2 variations, 'default' and 'custom link', but that in turn would add 4 a11y tests. I can imagine a situation in which we have multiple variations with snapshots and that would increase the testing suite exponentially. I'm wondering if we can do something like the following to reduce the maintenance burden:

```js
it('renders', async () => {
  const { container } = render(<TrendingCard />);
  
  expect(container.firstChild).toMatchSnapshot();

  await expect(container).toHaveNoAxeViolations();
});
```

Reply:
Thanks @SimonFinney I see what you mean now... and I question whether it would be helpful to group these checks into a single testing block. When you say this would "increase the testing block exponentially" are you referring to the testing blocks alone or something else? 

The blocks are helpful for group things together in a way that quickly scannable & follows Carbon's example (similar to: https://github.com/carbon-design-system/carbon/blob/master/packages/react/src/components/Loading/Loading-test.js#L20). It also makes things easier to upset in the future, because you only have one related test to update per block, in case something changes. Putting more things in one block feels like putting too much in a function, if that makes sense. 😅

Reply:
I certainly agree about having too much happening within testing blocks, but I feel like for snapshot tests it's less of an issue as they're validating the render output of the component, similar to a11y tests.

I suppose the point I'm getting at is that we're going to need to define these a11y tests for each variation, to account for all edge-cases. Looking at Carbon's `Loading` example, they specify a11y tests for the default example, but they are most likely missing coverage [here] (https://github.com/carbon-design-system/carbon/blob/master/packages/react/src/components/Loading/Loading-test.js#L97) and [here](https://github.com/carbon-design-system/carbon/blob/master/packages/react/src/components/Loading/Loading-test.js#L111). It seems like a missed opportunity to not reuse these variations given that they're intended to be an accurate representation of each of the component's states.

If you feel like it's outside the scope of this PR, I can create an issue to investigate how we might integrate these better, but I foresee a lot of repetition and maintenance burden ahead if we don't have some synchronicity between the 2.

Reply:
Thanks @SimonFinney that's a good point -- do you mind if we go the route of making a separate PR to explore this? 🙏  I'd like to see how you tackle it. Because with regards to maintenance burden, I'm having trouble seeing how putting things in the same block wouldn't make things harder to extend/revise in the future (as in, sacrificing that flexibility for the sake of avoiding a little repetition).